src/adapter/src/coord/sequencer/inner/copy_to.rs

```diff
@@ -0,0 +1,71 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+use std::str::FromStr;
+
+use http::Uri;
+
+use mz_repr::{Datum, RowArena};
+use mz_sql::plan::{self, CopyToPlan};
+
+use crate::coord::{Coordinator, TargetCluster};
+use crate::optimize::dataflows::{prep_scalar_expr, EvalTime, ExprPrepStyle};
+use crate::session::Session;
+use crate::{AdapterError, ExecuteResponse};
+
+impl Coordinator {
+    #[tracing::instrument(level = "debug", skip(self))]
+    pub(crate) async fn sequence_copy_to(
+        &mut self,
+        session: &Session,
+        plan: plan::CopyToPlan,
+        target_cluster: TargetCluster,
+    ) -> Result<ExecuteResponse, AdapterError> {
+        let CopyToPlan {
+            from: _from,
+            mut to,
+            connection: _connection,
+            format_params: _format_params,
+        } = plan;
+
+        let style = ExprPrepStyle::OneShot {
+            logical_time: EvalTime::Deferred,
+            session,
+            catalog_state: self.catalog.state(),
+        };
+
+        prep_scalar_expr(&mut to, style)?;
+        let temp_storage = RowArena::new();
+        let evaled = to.eval(&[], &temp_storage)?;
+
+        let to_url = match evaled {
```

Comment:
Should be able to use `evaled.unwrap_str()` here with `type_as` in planning. (Although also add a test with an explicit NULL and something that returns a non-str to make sure.)

Reply:
So, `unwrap_str()` is panicking will explicit NULL. Handling it separately (the `extract_secret` method was also doing that).

Reply:
Added a test for non-str